### PR TITLE
Handle destructor block

### PR DESCRIPTION
**Pull request by alkino**
* This block was still not handled add support for it

### CODE REVIEW

**Review by ohm314 — APPROVED** [review]
looks good! Can you also update `docs/language.rst`

File: src/codegen/codegen_c_visitor.cpp [hunk 1]
```diff
@@ -1251,6 +1251,9 @@ std::string CodegenCVisitor::compute_method_name(BlockType type) const {
     if (type == BlockType::Initial) {
         return method_name(naming::NRN_INIT_METHOD);
     }
+    if (type == BlockType::Destructor) {
+        return method_name(naming::NRN_DEST_METHOD);
```
Comment (pramodk) [inline src/codegen/codegen_c_visitor.cpp:1255]: Can I do one another `nitpick` ? short form `DEST` is interpreted as `destination`. I suggest `NRN_DESTRUCT_METHOD` to be more clear?

File: src/codegen/codegen_c_visitor.cpp [hunk 2]
```diff
@@ -3305,9 +3309,20 @@ void CodegenCVisitor::print_nrn_init(bool skip_init_check) {
 }
 
 
+void CodegenCVisitor::print_nrn_dest() {
```
Comment (pramodk) [inline src/codegen/codegen_c_visitor.cpp:3312]: same - `print_nrn_destructor`

File: src/codegen/codegen_naming.hpp [hunk 3]
```diff
@@ -125,6 +125,9 @@ static constexpr char FOR_NETCON_SEMANTIC[] = "fornetcon";
 /// nrn_init method in generated code
 static constexpr char NRN_INIT_METHOD[] = "nrn_init";
 
+/// nrn_destr method in generated code
+static constexpr char NRN_DEST_METHOD[] = "nrn_dest";
```
Comment (pramodk) [inline src/codegen/codegen_naming.hpp:129]: same

**Review by pramodk — APPROVED** [review]
LGTM but I recommend one renaming

**Review by pramodk — APPROVED** [review]

**Comment by pramodk** [comment]
@alkino : in the commit messages please add `fixes #issue-number` so that issues are linked & automatically closed from PR merges. (Issue #584)

**Pull request merged**